src/plugins/discover/public/application/main/utils/update_saved_search.ts

```diff
@@ -45,9 +47,12 @@ export function updateSavedSearch(
       .setField('query', services.data.query.queryString.getQuery())
       .setField('filter', services.data.query.filterManager.getFilters());
   } else if (state) {
+    const appFilters = state.filters ? cloneDeep(state.filters) : [];
+    const globalFilters = globalStateContainer.get()?.filters ?? [];
+
     savedSearch.searchSource
       .setField('query', state.query ?? undefined)
-      .setField('filter', state.filters ? cloneDeep(state.filters) : []);
+      .setField('filter', [...appFilters, ...globalFilters]);
```

Comment:
If `cloneDeep` was necessary here, should we use it for global filters too?

Reply:
I'm actually not sure why we need `cloneDeep` for `state.filters` here, although maybe @kertal has an idea? Could be some very legacy code. But since `globalStateContainer` is deserializing directly from the URL, I don't think we'll have the same risk of accidentally sharing object references here.